### PR TITLE
Registers package to sunpy template

### PR DESCRIPTION
**Pull request by CyclingNinja**

### CODE REVIEW

**Comment by CyclingNinja** [comment]
pre-commit.ci autofix

File: pyproject.toml [hunk 1]
```diff
@@ -1,10 +1,51 @@
...
+  "pytest",
+  "pytest-doctestplus",
+  "pytest-cov",
+  "pytest-xdist",
+  "pytest-mpl"
+]
+docs = [
+  "sphinx",
+  "sphinx-automodapi",
+  "sphinx-sphinx-theme",
```
Comment (Cadair) [inline pyproject.toml:35]:
```suggestion
  "sunpy-sphinx-theme",
```

File: pyproject.toml [hunk 2]
```diff
@@ -1,10 +1,51 @@
...
+description = "An interactive animation framework for matplotlib."
+requires-python = ">=3.10"
+readme = { file = "README.rst", content-type = "text/x-rst" }
+license = { file = "licenses/LICENSE.rst", content-type = "text/plain" }
+authors = [
+  { name = "The SunPy Developers", email = "sunpy@googlegroups.com" },
+]
+dependencies = [
+  "matplotlib>=3.5.0",
+  "numpy>=1.21.0",
```
Comment (Cadair) [inline pyproject.toml:19]:
```suggestion
  "numpy>=1.23.0",
```

File: tox.ini [hunk 3]
```diff
@@ -1,43 +1,64 @@
...
+    oldestdeps: pip install -r requirements-min.txt
+    pip freeze --all --no-input
 commands =
-    pip freeze
-    !figure: {env:PYTEST_COMMAND} {posargs}
-    figure: /bin/sh -c "mkdir -p ./figure_test_images; python -c 'import matplotlib as mpl; print(mpl.ft2font.__file__, mpl.ft2font.__freetype_version__, mpl.ft2font.__freetype_build_type__)' > ./figure_test_images/figure_version_info.txt"
-    figure: /bin/sh -c "pip freeze >> ./figure_test_images/figure_version_info.txt"
-    figure: /bin/sh -c "cat ./figure_test_images/figure_version_info.txt"
-    figure: python -c "import mpl_animators.tests.helpers as h; print(h.get_hash_library_name())"
-    figure: {env:PYTEST_COMMAND} --mpl --mpl-generate-summary=html --mpl-baseline-path=https://raw.githubusercontent.com/sunpy/sunpy-figure-tests/mpl-animators-main/figures/{envname}/ {posargs}
```
Comment (Cadair) [inline tox.ini:40]: We need this. Most of the tests in this repo are figure tests.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -20,7 +21,31 @@ concurrency:
...
 
 jobs:
+  core:
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@main
+    with:
+      submodules: false
+      coverage: codecov
+      toxdeps: tox-pypi-filter
+      envs: |
+        - linux: py312
```
Comment (Cadair) [inline .github/workflows/ci.yml:31]:
```suggestion
        - windows: py311
        - macos: py310
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -20,7 +21,31 @@ concurrency:
...
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+      - run: python -m pip install -U --user build
+      - run: python -m build . --sdist
+      - run: python -m pip install -U --user twine
+      - run: python -m twine check dist/*
+
   test:
```
Comment (Cadair) [inline .github/workflows/ci.yml:47]: Delete all of this section

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -30,28 +55,48 @@ jobs:
...
+      submodules: false
+      pytest: false
+      toxdeps: tox-pypi-filter
+      libraries: |
+        apt:
+          - graphviz
+      envs: |
+        - linux: build_docs
+
+  cron:
```
Comment (Cadair) [inline .github/workflows/ci.yml:76]: nuke this as well.

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -30,28 +55,48 @@ jobs:
...
+      coverage: codecov
+      toxdeps: tox-pypi-filter
+      envs: |
+        - linux: py311-devdeps
     secrets:
-      pypi_token: ${{ secrets.pypi_token }}
+      CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+
+  publish_pure:
+      needs: [test, docs]
```
Comment (Cadair) [inline .github/workflows/ci.yml:94]:
```suggestion
      needs: [core, docs]
```

**Review by Cadair — COMMENTED** [review]
Most of the CI is on circle CI because of *all* the figure tests, so the github actions config just tests non-figure on linux & mac

File: tox.ini (hunk 3 above)
Comment (CyclingNinja) [reply tox.ini:40]: then we don't need the pytest call thats replaced this do we?

Comment (Cadair) [reply tox.ini]: We will need a version of this with things like `figure: --mpl \` in the existing pytest command.

File: .github/workflows/ci.yml (hunk 4 above)
Comment (nabobalis) [reply .github/workflows/ci.yml:31]: Wont this also need linux?

File: .github/workflows/sub_package_update.yml [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Install Cruft
+        run: python -m pip install cruft
```
Comment (nabobalis) [inline .github/workflows/sub_package_update.yml:45]: Does this need to be Cadairs version?

File: .gitignore [hunk 9]
```diff
@@ -24,12 +24,13 @@ parts/
 sdist/
 var/
 wheels/
+share/python-wheels/
 *.egg-info/
 .installed.cfg
 *.egg
 MANIFEST
+mpl_animators/_version.py
 mpl_animators/version.py
```
Comment (nabobalis) [inline .gitignore:33]: Do we need both?

File: pyproject.toml [hunk 10]
```diff
@@ -1,10 +1,51 @@
 [build-system]
+requires = [
+  "setuptools>=62.1",
+  "setuptools_scm[toml]>=6.2",
+  "wheel",]
```
Comment (nabobalis) [inline pyproject.toml:5]:
```suggestion
  "wheel",
]
```

File: .github/workflows/sub_package_update.yml (hunk 8 above)
Comment (Cadair) [reply .github/workflows/sub_package_update.yml:45]: yes

**Review by nabobalis — APPROVED** [review]

**Pull request merged**